tools/metadata/src/helpers.ts

```diff
@@ -2,50 +2,58 @@ import * as ethers from 'ethers'
 import { ERC20_ABI, ERC20_BYTES32_ABI } from './constants'
 import { Arrayish } from 'ethers/utils'
 
-export function getContract(address: string, ABI: any, network = 'mainnet') {
-  const provider =
-    network === 'mainnet'
-      ? ethers.getDefaultProvider()
-      : ethers.getDefaultProvider(network.toLowerCase())
+export function getContract(
+  address: string,
+  ABI: any,
+  provider: ethers.providers.BaseProvider
+) {
   if (!address || !address.startsWith('0x')) {
     throw Error(`Invalid 'address' parameter '${address}'.`)
   }
-
   return new ethers.Contract(address, ABI, provider)
 }
 
-export async function getTokenName(tokenAddress: string, network: string) {
+export async function getTokenName(
+  tokenAddress: string,
+  provider: ethers.providers.BaseProvider
+) {
   if (!tokenAddress || !tokenAddress.startsWith('0x')) {
     throw Error(`Invalid 'tokenAddress' parameter`)
   }
 
-  return getContract(tokenAddress, ERC20_ABI, network)
+  return getContract(tokenAddress, ERC20_ABI, provider)
     .name()
     .catch(() =>
-      getContract(tokenAddress, ERC20_BYTES32_ABI, network)
+      getContract(tokenAddress, ERC20_BYTES32_ABI, provider)
         .name()
         .then((bytes32: Arrayish) => ethers.utils.parseBytes32String(bytes32))
     )
 }
 
-export async function getTokenSymbol(tokenAddress: string, network: string) {
+export async function getTokenSymbol(
+  tokenAddress: string,
+  provider: ethers.providers.BaseProvider
+) {
   if (!tokenAddress || !tokenAddress.startsWith('0x')) {
     throw Error(`Invalid 'tokenAddress' parameter`)
   }
 
-  return getContract(tokenAddress, ERC20_ABI, network)
+  return getContract(tokenAddress, ERC20_ABI, provider)
```

Comment:
same as above comment, currently this function doesn't need to be async as you are returning a promise, but to keep it consistent with the rest of the code I would keep the async and use await